sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -550,10 +550,23 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
         ng_netapi_receive(ng_ipv6_pid, rcv_pkt);
     }
     else {
+        /* forwarding packet */
+        ipv6_addr_t next_hop_ip;
+        /* determining next hop's IP address */
+        if (ipv6_addr_is_link_local(&(hdr->dst))) {
+            memcpy(&next_hop_ip, &(hdr->dst), sizeof(ipv6_addr_t));
+        }
+        else if (!ng_ndp_node_next_hop_ipv6_addr(&next_hop_ip, &iface,
```

Comment:
But this is not implemented yet anyway, but actually looking at the RFC I don't see a big difference for the next-hop determination itself.